src/api/server.rkt

```diff
@@ -120,29 +120,71 @@
 
 (define (manager-ask msg . args)
   (log "Asking manager: ~a, ~a.\n" msg args)
-  (match msg
-    ['wait (manager-ask-with-callback msg args)]
-    ['result (manager-ask-with-callback msg args)]
-    ['timeline (manager-ask-with-callback msg args)]
-    ['check (manager-ask-with-callback msg args)]
-    ['count (manager-ask-with-callback msg args)]
-    ['improve (manager-ask-with-callback msg args)]
-    [_ (place-channel-put manager (flatten (list msg args)))]))
+  (if manager
+      (match msg
+        ['wait (manager-ask-with-callback msg args)]
+        ['result (manager-ask-with-callback msg args)]
+        ['timeline (manager-ask-with-callback msg args)]
+        ['check (manager-ask-with-callback msg args)]
+        ['count (manager-ask-with-callback msg args)]
+        ['improve (manager-ask-with-callback msg args)]
+        [_ (place-channel-put manager (flatten (list msg args)))])
+      (match msg ; public commands
+        ['wait
+         (match-define (list hash-false job-id) args)
+         (hash-ref completed-work job-id)]
+        ['result
+         (match-define (list job-id) args)
+         (hash-ref completed-work job-id #f)]
+        ['timeline ; Blocks can't wait on a job well one in progress.
+         (match-define (list job-id) args)
+         (hash-ref completed-work job-id)]
+        ['check
+         (match-define (list job-id) args)
+         (if (hash-ref completed-work job-id #f) job-id #f)]
```

Comment:
```suggestion
         (and (hash-has-key? completed-work job-id) job-id)]
```

You might even just want `hash-has-key?` if the return value itself doesn't matter, just whether it's true or false.

Reply:
I will have to follow with another PR using  just `hash-hash-key?` as that triggers errors in `pages.rkt` and I don't wanna feature creep in this RP. Saving a link as todo.
https://github.com/herbie-fp/herbie/blob/20bc175e059f24ca79775c33d9b9d3e9237719bb/src/reports/pages.rkt#L21